tools/omnissm/functions/subscribe/main.go

```diff
@@ -68,28 +70,76 @@ func init() {
 }
 
 func handleConfigurationItemChange(detail manager.ConfigurationItemDetail) error {
-	managedId := identity.Hash(detail.ConfigurationItem.Name())
-	_, err, ok := mgr.Get(managedId)
+	entry, err, ok := mgr.Get(detail.ConfigurationItem.Name())
 	if err != nil {
 		return err
 	}
 	if !ok {
-		log.Info().Err(err).Msgf("instance not found: %#v", managedId)
+		log.Info().Err(err).Msgf("instance not found: %#v", entry.ManagedId)
 		return nil
 	}
-	switch detail.ConfigurationItem.ResourceType {
+	switch detail.ConfigurationItem.ConfigurationItemStatus {
 	case "ResourceDiscovered", "OK":
-		if err := mgr.Update(managedId, detail.ConfigurationItem); err != nil {
+		if err := mgr.Update(entry.ManagedId, detail.ConfigurationItem); err != nil {
 			return err
 		}
 	case "ResourceDeleted":
-		if err := mgr.Delete(managedId); err != nil {
+		if err := mgr.Delete(entry.ManagedId); err != nil {
 			return err
 		}
+		if ResourceDeletedSNSTopic != "" {
+			if err := publishResourceDeletedSNSTopic(entry, detail); err != nil {
+				return err
+			}
+		}
 	}
+
 	return nil
 }
 
+func publishResourceDeletedSNSTopic(managerInstance *store.RegistrationEntry, detail manager.ConfigurationItemDetail) error {
+	config := aws.NewConfig()
+	sess := session.New(config)
+	if CrossAccountRole != "" {
+		config.Credentials = stscreds.NewCredentials(sess, CrossAccountRole)
+	}
+	svc := sns.New(sess, config)
+
+	type resourceObj struct {
+		ResourceID   string
+		ManagerID    string
```

Comment:
I think this was supposed to be `ManagedID`

Reply:
fixed